notebooks/06-Custom-Extensions.ipynb

```diff
@@ -0,0 +1,577 @@
+    "ellipse_extension_uri = \"asdf://example.com/example-project/extensions/shapes-1.0.0\"\n",
+    "ellipse_tag = \"asdf://example.com/example-project/tags/ellipse-1.0.0\"\n",
+    "\n",
+    "ellipse_manifest_content = f\"\"\"\n",
+    "%YAML 1.1\n",
+    "---\n",
+    "id: {ellipse_manifest_uri}\n",
+    "extension_uri: {ellipse_extension_uri}\n",
+    "\n",
+    "title: Example Shape extension 1.0.0\n",
+    "description: Tags for example shape objects.\n",
+    "\n",
+    "tags:\n",
+    "  - tag_uri: {ellipse_tag}\n",
+    "    schema_uri: {ellipse_uri}\n",
+    "...\n",
+    "\"\"\"\n",
+    "\n",
+    "asdf.get_config().add_resource_mapping({ellipse_manifest_uri: ellipse_manifest_content})\n",
+    "\n",
+    "# check\n",
+    "schema = asdf.schema.load_schema(ellipse_manifest_uri)\n",
+    "asdf.schema.check_schema(schema)\n",
+    "asdf.schema.validate(ellipse_manifest_content, schema=schema)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create a `Converter`\n",
+    "\n",
+    "All converters should be constructed as subclasses of the abstract type `asdf.extension.Converter`,\n",
+    "which requires that you define two methods:\n",
+    "\n",
+    "1. `to_yaml_tree`: which converts a Python object into an ASDF tree.\n",
+    "2. `from_yaml_tree`: which converts an ASDF tree into a python object.\n",
+    "\n",
+    "Note that these methods can account for the type/tag of the objects attempting to be converted.\n",
+    "\n",
+    "Moreover your converter also needs to define to variables:\n",
+    "\n",
+    "1. `tags`: A list of the ASDF tags that this converter will be used by ASDF to read.\n",
+    "2. `types`: A list of the Python (object) types that this converter will be used by ASDF write.\n",
+    "\n",
+    "Note that these lists do not need to be indexed with respect to each other, and that in order for\n",
+    "the converter to actually be used by ASDF, at least one of the `tags` needs to be registered as a\n",
+    "resource with ASDF (usually via the entry point).\n",
+    "\n",
+    "An example converter for `Ellipse`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class EllipseConverter(asdf.extension.Converter):\n",
+    "    tags = [ellipse_tag]\n",
+    "    types = [Ellipse]\n",
+    "\n",
+    "    def to_yaml_tree(self, obj, tag, ctx):\n",
+    "        return {\n",
+    "            \"semi_major\": obj.semi_major,\n",
+    "            \"semi_minor\": obj.semi_minor,\n",
+    "        }\n",
+    "\n",
+    "    def from_yaml_tree(self, node, tag, ctx):\n",
+    "        return Ellipse(semi_major=node[\"semi_major\"], semi_minor=node[\"semi_minor\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that one will normally defer the import of the object to be created to inside `from_yaml_tree`\n",
```

Comment:
```suggestion
    "Note that, for performance of the entry points, one will normally defer the `import` of the object to be created\n",
```

Reply:
these performance improvements should be even more pronounced when [PEP690](https://peps.python.org/pep-0690/) is implemented